lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -108,11 +108,171 @@ class RsqrtOpConversion : public OpConversionPattern<tosa::RsqrtOp> {
   }
 };
 
+template <typename SrcOp>
+SmallVector<Value, 2>
+create_broadcast_op_if_needed(SrcOp &srcOp, ArrayRef<Value> operands,
+                              ConversionPatternRewriter &rewriter) {
+  // TOSA allows implicit broadcasting, so we need to insert broadcast_in_dim
+  // Ops if necessary
+  // clang-format off
+  // tosa.add tensor<8xf32>)     -> tensor<1x4x4x8xf32>     Broadcast dims = (3)
+  // tosa.add tensor<4x8xf32>)   -> tensor<1x4x4x8xf32>     Broadcast dims = (2, 3) 
+  // tosa.add tensor<4x4x8xf32>) -> tensor<1x4x4x8xf32>     Broadcast dims = (1, 2, 3)
+  // clang-format on
+
+  StringRef broadcastFuncName = "emitc::broadcast_in_dim";
```

Comment:
Here you're actually calling the core op. So as mentioned before, no need to add it to the tosa header.